### PR TITLE
fix: cocoapod cache issue

### PR DESCRIPTION
**Pull request by vinistevam**
## **Description**

This PR aims to fix an issue when run ios build when a dependency is updated. After analysing the latest cases reported facing issues with ios build I realised that cocoapod restore was overwriting the cocoa pod install in the `yarn setup` step. So, I moved the restore cocoa pod cache on top of the `yarn setup` step to take advantage of the cache and properly update the new dependencies. 

Pipeline error: https://app.bitrise.io/build/8ae57475-c72a-41ce-96f2-d3b6768b9d16
Error:
```
PhaseScriptExecution [CP]\ Check\ Pods\ Manifest.lock /Users/vagrant/git/ios/build/Build/Intermediates.noindex/MetaMask.build/Debug-iphonesimulator/MetaMask.build/Script-99E9B9D94E374E1799EFBA99.sh (in target 'MetaMask' from project 'MetaMask')
(1 failure)
error Command failed with exit code 65.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

## **Related issues**

Fixes: https://github.com/MetaMask/mobile-planning/issues/1487

## **Manual testing steps**

1. Update a dependency
2. Run smoke pipeline 
3. Install cocoapod should run and update the cache

### **Before**

Pipeline updating cocoa pod:
https://app.bitrise.io/build/8ae57475-c72a-41ce-96f2-d3b6768b9d16

### **After**

Pipeline updating cocoa pod adding back the cocopod install step:
https://app.bitrise.io/build/8631a2cf-89f7-4025-964f-4c8bfeb4e28c

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by vinistevam** [comment]
pipeline: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/2e6a9911-3aea-41ca-b50e-64826468993f

**Review by OGPoyraz — DISMISSED** [review]

File: bitrise.yml [hunk 1]
```diff
@@ -589,6 +589,9 @@ workflows:
                 envman add --key BREW_OPT_APPLESIMUTILS --value "/usr/local/opt/applesimutils"
                 brew tap wix/brew
           title: Set Env Path for caching deps
+      - cocoapods-install@2:
```
Comment (sethkfman) [inline bitrise.yml:592]: This is already done in the `yarn setup` step. Can you move the restore cache to earlier in the pipeline it doesn't duplicate the steps?

Comment (vinistevam) [reply bitrise.yml]: Make sense, moved the cocoa pod restore on top of the `yarn setup`.

**Review by sethkfman — APPROVED** [review]
LGTM

**Pull request merged**